cfme/tests/cloud/test_providers.py

```diff
@@ -597,11 +598,38 @@ def test_refresh_with_empty_iot_hub_azure(request, provider, setup_provider):
     assert result.validate(wait="60s")
 
 
+@pytest.fixture
+def azure_available_regions(provider):
+    def _func():
+        view = navigate_to(AzureProvider, "Add")
+        view.browser.refresh()
+        # prefill the provider type to enable regions dropdown
+        view.fill({'prov_type': provider.type.capitalize()})
+        return [opt.text for opt in view.region.all_options][1:]  # first option is '<Choose>'
+    return _func
+
+
+@pytest.fixture
+def azure_available_regions_backend(appliance):
+    regions_rb = appliance.ssh_client.run_command(
+        "cat /opt/rh/cfme-gemset/bundler/gems/cfme-providers-azure-*/"
+        "app/models/manageiq/providers/azure/regions.rb")
+    c = [x.group(1) for x in re.finditer(r':description => _\(\"([ \w]*)\"\)',
+                                         regions_rb.output,
+                                         flags=re.MULTILINE)]
```

Comment:
These regions should be available in the appliance advanced_settings property, which reads the advanced settings yaml. There are also appliance methods to enable/disable these regions.  I think that's a much better way to read them than trying to parse them directly from the ruby file. This test will be unreasonably sensitive to changes in the MIQ source.

Have you talked to @mmojzis about this testing? We hit the same thing in EC2 - MIQ only updates the included regions periodically, and sometimes there is a significant lag between a new region being added to the cloud service, and MIQ making it available.

These could be parametrized for azure and ec2, comparing the regions available from the provider API directly (wrapanapi), to the advanced_settings regions, to the UI listed regions.

Reply:
I would say it would be the best to get regions in wrapanapi? (https://github.com/Azure/azure-cli/issues/1520) And then have parametrized test? I have it here (https://github.com/ManageIQ/integration_tests/pull/9706/files#diff-4f10bc4cf82a0f08488c8079d4cd0085R1347) I just compare regions from wrapanapi with regions that are available during cloud provider add form.